### DISCUSSION TITLE
How do I programmatically define an argument to be mandatory

### DISCUSSION THREAD

**Discussion by Xnyle** [graphql-java/graphql-java#2215] [Q&A] [answered]
I couldn't find any an example / doc how to define a query arg to be mandatory when programatically creating a schema.

queryTypeBuiler.field(newFieldDefinition()
					.name("findbyParamsOrWhatever")
					.argument(new GraphQLArgument("limit", GraphQLInt))

How do I define limit to be mandatory or not?
Doc says if you define a type and no default value it becomes mandatory but in my mind I just did that, define a type and no default value?

There also is 

GraphQLArgument("limit", "limit", GraphQLInt, null, InputValueDefinition.newInputValueDefinition().type(GraphQLInt));

But no idea what .type() wants to have as parameter.

**Comment by bbakerman** [comment, marked as answer]
```
new GraphQLArgument("limit", GraphqlNonNull.nonNull(GraphQLInt))
```

This is a type that is non nullable (with no default) and hence requires a value to be valid

in SDl is the equivalent of `limit : Int!`
